config/routes.rb

```diff
@@ -1,3 +1,6 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  get "/guests/:id", to: "guests#show"
+
+  post "/guests/:id/rooms", to: "room#create"
```

Comment:
Great job setting up the guest's show route correctly with the right controller and action. The post route to `"/guests/:id/rooms"` is also correct. The controller and action that you have for this URI path is what I want to focus on here.

Let's say we have this route: `post "/rooms"`, this would also go to the `RoomsController` and the `create` but we never do anything inside of the `create` action to determine what route was actually used to hit the action.

Was it `post "/guests/:id/rooms"` or `post "/rooms"`? (I know we don't have a create rooms user story right now, but we have to think about that functionality possibly being added later on)

Right now your `RoomsController` `create` action looks like this:

```ruby
def create
   guest_room = GuestRoom.new(guest_id: params[:id], room_id: params[:room_id])
   guest_room.save
   redirect_to "/guests/#{params[:id]}"
end
```

If it was a `post "/rooms"` that hit this action, this logic would fail because there would be no `:room_id`. Instead of going into how you could modify this method to work regardless of which of the path triggered the action, I would recommend you to use a join controller like `GuestRoomsController` and in there have your `create` action.